alma/controllers/hook/StateHookController.php

```diff
@@ -43,38 +43,91 @@ final class StateHookController extends AdminHookController
     use OrderDataTrait;
 
     /**
+     * Execute query action on change state
      * @param $params
      *
      * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
      */
     public function run($params)
     {
-        if (!Settings::isRefundEnabledByState()) {
-            return;
-        }
-
         $order = new Order($params['id_order']);
         $newStatus = $params['newOrderStatus'];
         if ($order->module !== 'alma' || !$order_payment = $this->getOrderPaymentOrFail($order)) {
             return;
         }
-
+        $alma = ClientHelper::defaultInstance();
+        if (!$alma) {
+            return;
+        }
         $id_payment = $order_payment->transaction_id;
-        if ($newStatus->id == Settings::getRefundState()) {
-            $alma = ClientHelper::defaultInstance();
-            if (!$alma) {
+
+        switch ($newStatus->id) {
+            case Settings::getRefundState():
+                $this->validateState(Settings::isRefundEnabledByState());
+                $this->queryRefund($alma, $id_payment, $order);
+                break;
+            case Settings::getPaymentTriggerState():
```

Comment:
Are you sure `getPaymentTriggerState()` cannot  return the same `int` than `getRefundState()` ?

Reply:
But this config can be changed by the merchant ?
If he decide to set the same state for trigger and refund... what will be happen ?

Reply:
so, we have to manage it in our code

Reply:
you are right, I thank switch case will throws an Error.... but in fact, it doesn't

Reply:
No, he'll throw the first one, right? Can I add a configuration check to prevent both having the same value? That would make more sense I think.